Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -25,33 +25,24 @@ OrderImbalanceIndicatorModel::SubscriptionResult
 void LocalOrderImbalanceIndicatorModel::insert(
     const OrderImbalance& imbalance) {
   m_imbalances.push_back(imbalance);
-  if(!m_signals.empty()) {
-    update_signals(0, m_signals.size(), m_signals.size() - 1, imbalance);
+  auto end_index = m_signals.size();
+  auto swap_index = end_index;
+  for(auto i = std::size_t(0); i < end_index && i != swap_index; ++i) {
+    if(m_signals[i].m_imbalance_signal.num_slots() != 0) {
+      if(m_signals[i].m_start_time <= imbalance.m_timestamp &&
+          imbalance.m_timestamp <= m_signals[i].m_end_time) {
+        m_signals[i].m_imbalance_signal(imbalance);
+      }
+    } else {
+      std::swap(m_signals[i], m_signals[--swap_index]);
+      --i;
+    }
   }
+  m_signals.erase(m_signals.begin() + swap_index,
+    m_signals.begin() + end_index);
```

Comment:
When will ```end_index``` ever be different from ```m_signals.size()```?

Reply:
When a called signal results in a call to subscribe and a new signal gets inserted.

Reply:
Good.